pootle/apps/pootle_store/views.py

```diff
@@ -753,8 +761,20 @@ def get_edit_unit(request, unit):
     alt_src_langs = get_alt_src_langs(request, user, translation_project)
     project = translation_project.project
 
-    # Retrieve the unit top priority, if any.
-    priority = unit.vfolders.aggregate(priority=Max('priority'))['priority']
+    priority = None
+    vfolder_pk = request.GET.get('vfolder', '')
+
+    if vfolder_pk:
+        # If we are translating a virtual folder, then display its priority.
+        try:
+            vfolder = VirtualFolder.objects.get(pk=vfolder_pk)
+            priority = vfolder.priority
+        except Exception:
+            pass
+
+    if priority is None:
+        # If we have no priority, then retrieve the unit top priority, if any.
+        priority = unit.vfolders.aggregate(priority=Max('priority'))['priority']
```

Comment:
Depending on the meaning of the exception handler above, you might be able to move this up to the `except` block.